src/api/v2/db/ldap/directory-dao.js

```diff
@@ -11,26 +11,29 @@ import { getClient } from './connection';
  * @returns {string} string representing search filter for ldap query
  */
 const mapQuery = (endpointQuery) => {
-  const keyMap = {
-    fuzzyName: 'cn',
-    lastName: 'sn',
-    firstName: 'givenName',
-    primaryAffiliation: 'osuPrimaryAffiliation',
-    onid: 'uid',
-    emailAddress: 'mail',
-    officePhoneNumber: 'telephoneNumber',
-    alternatePhoneNumber: 'osuAltPhoneNumber',
-    faxNumber: 'facsimileTelephoneNumber',
-    phoneNumber: 'telephoneNumber',
-    officeAddress: 'osuOfficeAddress',
-    department: 'osuDepartment',
+  const keyMap = (key) => {
+    switch (key) {
+      case 'filter[fullName][fuzzy]': return 'cn';
+      case 'filter[lastName]': return 'sn';
+      case 'filter[firstName]': return 'givenName';
+      case 'filter[primaryAffiliation]': return 'osuPrimaryAffiliation';
+      case 'filter[onid]': return 'uid';
+      case 'filter[emailAddress]': return 'mail';
+      case 'filter[officePhoneNumber][fuzzy]': return 'telephoneNumber';
+      case 'filter[alternatePhoneNumber][fuzzy]': return 'osuAltPhoneNumber';
+      case 'filter[faxNumber][fuzzy]': return 'facsimileTelephoneNumber';
+      case 'filter[phoneNumber][fuzzy]': return 'telephoneNumber';
+      case 'filter[officeAddress][fuzzy]': return 'osuOfficeAddress';
+      case 'filter[department]': return 'osuDepartment';
+      default: return undefined;
+    }
```

Comment:
Just curious, did you migrate to a switch statement because of the `[]` in the keys? If so, you could still use an object by surrounding the keys in quotes. I think this might be better just to cut down on the repeated `case` and `return`s unless there was another reason for using this.

Reply:
That's a good point. I'll change it back.